geomstats/geometry/pullback_metric.py

```diff
@@ -111,6 +113,53 @@ def pickable_inner_product(i, j):
         metric_mat = gs.reshape(gs.array(out), (-1, self.dim, self.dim))
         return metric_mat[0] if base_point.ndim == 1 else metric_mat
 
+    def _compute_hessian_immersion(self):
```

Comment:
Have you noticed any real gain of efficiency? (In my view it may avoid some overhead on defining functions, but not much more other than that.) Not sure if `_compute_hessian` is a good name though (since the computation is only being done when `base_point` is passed.